### PR TITLE
Python: Renames request settings to prompt execution settings

### LINKED ISSUE #4676: Python: Rename request settings to execution settings.
Per #4595, rename request settings to execution settings.

### PR DESCRIPTION
**Pull request by juliomenendez**
### Motivation and Context

Following the C# implementation, renames all classes and variables of `*RequestSettings` to `*PromptExecutionSettings`.

Fixes #4676 

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [X] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

### CODE REVIEW

File: python/semantic_kernel/connectors/ai/chat_completion_client_base.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ async def complete_chat(
         Arguments:
             messages {List[ChatMessage]} -- A list of chat messages, that can be rendered into a
                 set of messages, from system, user, assistant and function.
-            settings {AIRequestSettings} -- Settings for the request.
+            settings {AIPromptExecutionSettings} -- Settings for the request.
```
Comment (moonbox3) [inline python/semantic_kernel/connectors/ai/chat_completion_client_base.py:25]: Should this be `PromptExecutionSettings`?

File: python/semantic_kernel/connectors/ai/chat_completion_client_base.py [hunk 2]
```diff
@@ -43,7 +43,7 @@ async def complete_chat_stream(
...
```
Comment (moonbox3) [inline python/semantic_kernel/connectors/ai/chat_completion_client_base.py:46]: Should this be PromptExecutionSettings?

File: python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py [hunk 3]
```diff
@@ -50,24 +50,24 @@ async def complete_chat(
         Arguments:
             messages {List[ChatMessage]} -- A list of chat messages, that can be rendered into a
                 set of messages, from system, user, assistant and function.
-            settings {AIRequestSettings} -- Settings for the request.
+            settings {AIPromptExecutionSettings} -- Settings for the request.
```
Comment (moonbox3) [inline python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py:53]: Should this be PromptExecutionSettings?

File: python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py [hunk 4]
```diff
@@ -96,26 +96,26 @@ async def complete_chat_stream(
...
+        prompt_execution_settings: OllamaChatPromptExecutionSettings,
         **kwargs,
     ) -> Union[str, List[str]]:
         """
         This is the method that is called from the kernel to get a response from a text-optimized LLM.
 
         Arguments:
             prompt {str} -- The prompt to send to the LLM.
-            settings {AIRequestSettings} -- Settings for the request.
+            settings {AIPromptExecutionSettings} -- Settings for the request.
```
Comment (moonbox3) [inline python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py:107]: Should this be PromptExecutionSettings?

File: python/semantic_kernel/connectors/ai/ollama/services/ollama_text_completion.py [hunk 5]
```diff
@@ -37,31 +37,31 @@ class OllamaTextCompletion(TextCompletionClientBase, AIServiceClientBase):
...
+        prompt_execution_settings: OllamaTextPromptExecutionSettings,
         **kwargs,
     ) -> Union[str, List[str]]:
         """
         This is the method that is called from the kernel to get a response from a text-optimized LLM.
 
         Arguments:
             prompt {str} -- The prompt to send to the LLM.
-            settings {AIRequestSettings} -- Settings for the request.
+            settings {AIPromptExecutionSettings} -- Settings for the request.
```
Comment (moonbox3) [inline python/semantic_kernel/connectors/ai/ollama/services/ollama_text_completion.py:48]: My last comment, let's do a global search and replace for this -> PromptExecutionSettings. :)

File: python/semantic_kernel/connectors/ai/prompt_execution_settings.py [hunk 6]
```diff
@@ -5,20 +5,23 @@
 from semantic_kernel.kernel_pydantic import KernelBaseModel
 
 
-class AIRequestSettings(KernelBaseModel):
-    """Base class for AI request settings.
+class PromptExecutionSettings(KernelBaseModel):
+    """Base class for prompt execution settings.
 
-    Can be used by itself or as a base class for other request settings. The methods are used to create specific request settings objects based on the keys in the extension_data field, this way you can create a generic AIRequestSettings object in your application, which get's mapped into the keys of the request settings that each services returns by using the service.get_request_settings() method.
+    Can be used by itself or as a base class for other prompt execution settings. The methods are used to create
```
Comment (moonbox3) [inline python/semantic_kernel/connectors/ai/prompt_execution_settings.py:11]: Extra white space added?

**Review by moonbox3 — COMMENTED** [review]
Looks good! A few small comments.

File: python/semantic_kernel/connectors/ai/prompt_execution_settings.py (hunk 6 above)
Comment (juliomenendez) [reply python/semantic_kernel/connectors/ai/prompt_execution_settings.py:11]: I manually broke the line into multiple lines to fit in 120 characters. I can revert it if we think docstrings shouldn't follow that rule.

File: python/semantic_kernel/connectors/ai/chat_completion_client_base.py (hunk 1 above)
Comment (juliomenendez) [reply python/semantic_kernel/connectors/ai/chat_completion_client_base.py]: Fixed

File: python/semantic_kernel/connectors/ai/chat_completion_client_base.py (hunk 2 above)
Comment (juliomenendez) [reply python/semantic_kernel/connectors/ai/chat_completion_client_base.py]: Fixed

File: python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py (hunk 3 above)
Comment (juliomenendez) [reply python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py]: Fixed

File: python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py (hunk 4 above)
Comment (juliomenendez) [reply python/semantic_kernel/connectors/ai/ollama/services/ollama_chat_completion.py]: Fixed

File: python/semantic_kernel/connectors/ai/prompt_execution_settings.py (hunk 6 above)
Comment (moonbox3) [reply python/semantic_kernel/connectors/ai/prompt_execution_settings.py:11]: Oh, it was a single line before, that's why there's so much white space. No worries, I like cutting on 120 chars max.

**Review by moonbox3 — APPROVED** [review]

File: python/semantic_kernel/connectors/ai/ollama/services/ollama_text_completion.py (hunk 5 above)
Comment (juliomenendez) [reply python/semantic_kernel/connectors/ai/ollama/services/ollama_text_completion.py]: Fixed

**Review by eavanvalkenburg — APPROVED** [review]
LGTM

**Review by moonbox3 — APPROVED** [review]

**Review by moonbox3 — APPROVED** [review]

File: python/semantic_kernel/connectors/ai/open_ai/services/azure_chat_completion.py [hunk 7]
```diff
@@ -273,9 +273,9 @@ def from_dict(cls, settings: Dict[str, str]) -> "AzureChatCompletion":
             default_headers=settings.get("default_headers"),
         )
 
-    def get_request_settings_class(self) -> "AIRequestSettings":
+    def get_request_settings_class(self) -> "PromptExecutionSettings":
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/connectors/ai/open_ai/services/azure_chat_completion.py:276]: this needs to be renamed!

File: python/semantic_kernel/connectors/ai/open_ai/services/open_ai_text_completion_base.py [hunk 8]
```diff
@@ -9,44 +9,44 @@
...
+    from semantic_kernel.connectors.ai.open_ai.prompt_execution_settings.open_ai_prompt_execution_settings import (
+        OpenAIPromptExecutionSettings,
     )
 
 logger: logging.Logger = logging.getLogger(__name__)
 
 
 class OpenAITextCompletionBase(OpenAIHandler, TextCompletionClientBase):
-    def get_request_settings_class(self) -> "AIRequestSettings":
+    def get_request_settings_class(self) -> "PromptExecutionSettings":
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/connectors/ai/open_ai/services/open_ai_text_completion_base.py:30]: this one as well

File: python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py [hunk 9]
```diff
@@ -35,22 +35,22 @@
 class OpenAIChatCompletionBase(OpenAIHandler, ChatCompletionClientBase):
     """OpenAI Chat completion class."""
 
-    def get_request_settings_class(self) -> "AIRequestSettings":
+    def get_request_settings_class(self) -> "PromptExecutionSettings":
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py:38]: and here

File: python/semantic_kernel/connectors/ai/open_ai/services/azure_chat_completion.py (hunk 7 above)
Comment (juliomenendez) [reply python/semantic_kernel/connectors/ai/open_ai/services/azure_chat_completion.py]: This doesn't seem to have any effect on the logic so I'm going to leave it as-is for now. I can rename it in a follow-up PR if needed.

File: python/semantic_kernel/connectors/ai/open_ai/services/open_ai_text_completion_base.py (hunk 8 above)
Comment (juliomenendez) [reply python/semantic_kernel/connectors/ai/open_ai/services/open_ai_text_completion_base.py]: This doesn't seem to have any effect on the logic so I'm going to leave it as-is for now. I can rename it in a follow-up PR if needed.

File: python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py (hunk 9 above)
Comment (juliomenendez) [reply python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py]: This doesn't seem to have any effect on the logic so I'm going to leave it as-is for now. I can rename it in a follow-up PR if needed.

**Review by eavanvalkenburg — APPROVED** [review]
LGTM

**Pull request merged**